app/models/recipe.rb

```diff
@@ -0,0 +1,18 @@
+class Recipe < ApplicationRecord
+  belongs_to :user
+  has_many :recipe_foods, dependent: :destroy
+  validates :name, presence: true
+  validates :preparation_time, numericality: { greater_than_or_equal_to: 0 }
+  validates :cooking_time, numericality: { greater_than_or_equal_to: 0 }
+  validates :user_id, presence: true
+
```

Comment:
- [ ] The Recipe model belongs to a User (belongs_to :user) and should have a many-to-many association with the Food model through the RecipeFood join table. This is achieved by using `has_many :recipe_foods` and `has_many :foods, through: :recipe_foods`. KIndly fix this many-to-many association. 

```ruby
class Recipe < ApplicationRecord
  belongs_to :user
  has_many :recipe_foods
  has_many :foods, through: :recipe_foods
end
```

Reply:
well resolved! ✔️